File: online-test-intellij/test-service/src/main/java/com/onlinetest/testservice/service/TestService.java
package com.onlinetest.testservice.service;

import com.onlinetest.testservice.dto.Response;
import com.onlinetest.testservice.entity.Test;
import com.onlinetest.testservice.exceptions.TestNotFoundException;

import java.util.List;

public interface TestService {
    public Test addTest(Test test);
    public Test updateTest(Test test) throws TestNotFoundException;
    public Test getTestById(Integer id) throws TestNotFoundException;
    public String deleteTestById(Integer id) throws TestNotFoundException;
    public Integer getScore(List<Response> responses);
    public List<Test> getAllTests() throws TestNotFoundException;
}

File: src/app/test.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Test} from "./Test";
import {Observable} from "rxjs";
@Injectable({
  providedIn: 'root'
})
export class TestService {
  private testUrl: string = "/test-service/tests/";



  constructor(private http: HttpClient) { }

  getAllTests(): Observable<Test[]> {
    return this.http.get<Test[]>('http://localhost:8081/tests/');
  }
  addTest(test: Test): Observable<Test> {
    return this.http.post<Test>(this.testUrl, test);
  }
  getTestById(id: number): Observable<Test> {
    return this.http.get<Test>(this.testUrl+"/"+id);
  }
  getScore(responses :{questionId: number, chosenOption: number}[]): Observable<number>{
    return this.http.post<number>(this.testUrl+"getScore", responses);
  }
  deletetest(id: number): Observable<void>{
  return this.http.delete<void>(this.testUrl+id);
  }


}
